File: 268. Missing Number/268._Missing_Number_1.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        xor = 0
        for i in range(len(nums)):
            xor = xor ^ i ^ nums[i]
        
        return xor ^ len(nums)

File: 268. Missing Number/268._Missing_Number_1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        if (nums.empty() || nums.size() == 0)
            return 0;
        int XOR = 0, i = 0;
        for (i = 0; i < nums.size(); i++) {
            XOR = XOR ^ i ^ nums[i];
        }
        return XOR ^ i;
    }
};